org.eclipse.jdt.core/model/org/eclipse/jdt/core/IMethod.java

```diff
@@ -255,6 +255,14 @@ public interface IMethod extends IMember, IAnnotatable {
  * <li>it is <code>static</code> and <code>public</code></li>
  * <li>it defines one parameter whose type's simple name is <code>String[]</code></li>
  * </ul>
+ * Starting with Java 21 there is "preview feature" that allows <a href="https://openjdk.org/jeps/445">instance main methods</a>.
```

Comment:
We should keep this API/contract untouched and create a new API with @noreference, so that existing clients are not broken.